### ISSUE TITLE
Reproducing Segmentation Results

### ISSUE THREAD

**Issue by khalilsarwari** [xu-ji/IIC#64]
Hi!

I am trying to reproduce the results on POTSDAM using the example command you provided:

```
export CUDA_VISIBLE_DEVICES=0,1,2,3 && nohup python -m code.scripts.segmentation.segmentation_twohead --mode IID --dataset Potsdam --dataset_root /home/khalil/Documents/IIC/datasets/POTSDAM --model_ind 544 --arch SegmentationNet10aTwoHead --num_epochs 4800 --lr 0.000001 --lamb_A 1.0 --lamb_B 1.0 --num_sub_heads 1 --batch_sz 60 --num_dataloaders 1 --output_k_A 36 --output_k_B 6 --gt_k 6 --input_sz 200 --half_T_side_sparse_min 0 --half_T_side_sparse_max 0 --half_T_side_dense 5  --include_rgb --no_sobel --jitter_brightness 0.1 --jitter_contrast 0.1 --jitter_saturation 0.1 --jitter_hue 0.1 --use_uncollapsed_loss --batchnorm_track > gnoded2_gpu0123_m544_r1.out &
```

I keep running into variations on the following output:
`only got 0.44026533 4th, restarting`

1.) I've tried multiple times to see if I get lucky enough to pass this catch, but every time the accuracy is too low. I understand that the initialization probably matters a lot, so is it better to restart than to continue training? Or should I comment out this check and just train for longer? How many times should I expect to have to rerun the script to be able to reproduce the results?

2.) Did you by any chance experiment with doing segmentation with more classes? The results in the paper seem to be for relatively few ground truth classes, and I was wondering if there is any hope in using this method for a dataset with 16+ ground truth classes.

Thanks for your time, and this is really superb work!

**Comment by xu-ji** [comment]
I think you have an old version of the code, that check was removed a while back.
I regret only giving point estimates in the results tables, but you should be able to run the code a few times and get close to the reported performance. In fact 45.4 was reported for that model, so 44.0 is reasonably close.

Theoretically there is no reason why it shouldn't scale to more ground truth classes. The paper was a proof of concept so we opted for fairly simple dataset settings.

**Comment by limai1990** [comment]
Hi @khalilsarwari ,

Is your segmentation score (0.44026533) reproduced on POTSDAM stable? Can different initializations get similar scores? Will this score be obtained in the first five epochs of the experiment?

I got very fluctuating segmentation scores under different initializations.

Thanks,
Li

**Comment by khalilsarwari** [comment]
If I remember correctly it did fluctuate quite a bit; I don't remember the epoch for which the score is obtained by unfortunately.
